packages/react-components/src/components/InAppMessages/InAppMessage.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import cx from 'clsx';
+
+import {
+  InAppMessageHeader,
+  InAppMessageHeaderProps,
+} from './InAppMessageHeader';
+import { InAppMessageImage, InAppMessageImageProps } from './InAppMessageImage';
+import {
+  InAppMessageFooter,
+  InAppMessageFooterProps,
+} from './InAppMessageFooter';
+import { KeyCodes } from '../../utils/keyCodes';
+
+import styles from './InAppMessage.module.scss';
+
+const baseClass = 'in-app-message';
+
+export interface InAppMessageProps {
+  className?: string;
+  image?: InAppMessageImageProps;
+  header?: InAppMessageHeaderProps;
+  footer?: InAppMessageFooterProps;
```

Comment:
IMO in this case, the composition is better than the configuration. Instead of providing all of those nested props objects, why don't you just provide an instance of the image/header/footer component in the form of ReactElement? That way consumer will provide the required props for those subcomponents on his own. 

Additionally, React composition allows reducing the number of unnecessary re-renders because you are not creating new object references on each render.  The sent thing also is that typically UI library built around composition is easier to use and extend in more advanced cases.

So here the types will look like this:
```suggestion
  image?: React.ReactElement;
  header?: React.ReactElement;
  footer?: React.ReactElement;
```

And later on, render:
```ts
  return (
    <div
      data-testid={`${baseClass}-overlay`}
      className={styles[`${baseClass}__overlay`]}
      onMouseDown={handleOnOverlayClick}
    >
      <div className={styles[`${baseClass}__container`]}>
        {header}
        <div className={mergedClassNames}>
          <div className={styles[`${baseClass}__content`]}>
            {image}
            <div
              className={cx(styles[`${baseClass}__content__body`], {
                [styles[`${baseClass}__content__body__with-footer`]]: !!footer,
              })}
            >
              {children}
            </div>
          </div>
          {footer}
        </div>
      </div>
    </div>
  );
```

WDYT? 🤔